File: source/dotnet-petclinic-rest/PetClinic.Application/Dtos/VisitDTO.cs
using System;
using System.Collections.Generic;
using AutoMapper;
using Intent.RoslynWeaver.Attributes;
using PetClinic.Application.Common.Mappings;
using PetClinic.Domain.Entities;

[assembly: DefaultIntentManaged(Mode.Fully)]
[assembly: IntentTemplate("Intent.Application.Dtos.DtoModel", Version = "1.0")]

namespace PetClinic.Application.Dtos
{

    public class VisitDTO : IMapFrom<Visit>
    {
        public VisitDTO()
        {
        }

        public static VisitDTO Create(
            int id,
            DateTime visitDate,
            string description,
            PetDTO pet,
            int ownerId)
        {
            return new VisitDTO
            {
                Id = id,
                VisitDate = visitDate,
                Description = description,
                Pet = pet,
                OwnerId = ownerId,
            };
        }

        public int Id { get; set; }

        public DateTime VisitDate { get; set; }

        public string Description { get; set; }

        public PetDTO Pet { get; set; }

        public int OwnerId { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<Visit, VisitDTO>()
                .ForMember(d => d.OwnerId, opt => opt.MapFrom(src => src.Pet.Owner.Id));
        }
    }
}

File: source/nestjs-petclinic-rest/src/services/dto/visit.dto.ts
import { PetDTO } from './pet.dto';
import { Visit } from './../../domain/entities/visit.entity';
import { ApiProperty } from '@nestjs/swagger';
import { Type } from 'class-transformer';

export class VisitDTO
{
  @ApiProperty()
  id: number;

  @ApiProperty()
  @Type(() => Date)
  visitDate: Date;

  @ApiProperty()
  description: string;

  @ApiProperty()
  pet: PetDTO;

  @ApiProperty()
  ownerId: number;

  static fromVisit(visit: Visit) {
    if (visit == null) {
        return null;
    }
    const dto = new VisitDTO();
    dto.id = visit.id;
    dto.visitDate = visit.visitDate;
    dto.description = visit.description;
    dto.pet = PetDTO.fromPet(visit.pet);
    dto.ownerId = visit.pet?.owner?.id;
    return dto;
  }

  static requiredRelations: string[] = ["pet", "pet.petType", "pet.owner", "pet.visits"];
}